### ISSUE TITLE
VerifyProvider: provider name not being used when verifying with broker

### ISSUE THREAD

**Issue by jasonltaylor** [pact-foundation/pact-go#256]
## Software versions

* **OS**: Ubuntu Linux
* **Consumer Pact library**: Pact Go CLI v2.0.0-beta.17
* **Provider Pact library**: Pact Go CLI v2.0.0-beta.17
* **Golang Version**: go version go1.19.2 linux/amd64
* **Golang environment**: 
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/dev/.cache/go-build"
GOENV="/home/dev/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/dev/go/pkg/mod"
GONOPROXY="github.hpe.com"
GONOSUMDB="github.hpe.com"
GOOS="linux"
GOPATH="/home/dev/go"
GOPRIVATE="github.hpe.com"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19.2"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/dev/ws/sc-authz/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4109696144=/tmp/go-build -gno-record-gcc-switches"

## Expected behaviour

When verifying a provider contract, the provider name should be properly used to check for consumer contracts.

## Actual behaviour

Provider name is set to "provider" which results in no contracts found.

## Steps to reproduce

This should be reproducible by adding BrokerURL integration to the existing gRPC provider example:
https://github.com/pact-foundation/pact-go/blob/2.x.x/examples/grpc/grpc_provider_test.go

## Relevent log files

High level error when performing provider verification - can see that the path being used is 'provider/provider' instead of 'provider/<provider name>':

```
[2022-12-08T19:55:52.829Z] 2022-12-08T19:55:49.588545Z  INFO ThreadId(03) pact_verifier::pact_broker: Fetching path '/' from pact broker

[2022-12-08T19:55:52.829Z] 2022-12-08T19:55:49.611054Z  INFO ThreadId(03) pact_verifier::pact_broker: Fetching path '/pacts/provider/provider/for-verification' from pact broker

[2022-12-08T19:55:52.829Z] 2022-12-08T19:55:49.617698Z  WARN ThreadId(03) pact_verifier: Ignoring no pacts error - Could not load pacts from the pact broker 
```

More detailed logging enabled:

```
2022-12-08T22:17:00.954892Z DEBUG ThreadId(01) pact_ffi::verifier::handle: Pact source to verify = PactBrokerWithDynamicConfiguration(https://redacted.com/, provider_name='provider', enable_pending=false, include_wip_since=None, provider_tags=[], provider_branch=Some(""), consumer_version_selectors='[ConsumerVersionSelector { consumer: None, tag: None, fallback_tag: None, latest: None, deployed_or_released: None, deployed: None, released: None, environment: None, main_branch: Some(true), branch: None, matching_branch: None }], auth=None')
```

**Comment by mefellows** [comment]
Thanks for this. I believe this should be fixed in the latest via #255. Would you mind please checking the latest HEAD on the 2.x.x branch? (I'll release it once a few more things are tidied up)

**Comment by jasonltaylor** [comment]
Verified fixed in latest 2.x.x branch - tx!
